--- origsrc/seed-2.27.92/libseed/seed-importer.c	2009-09-03 15:19:06.000000000 -0500
+++ src/seed-2.27.92/libseed/seed-importer.c	2009-09-08 01:46:29.731630800 -0500
@@ -692,7 +692,7 @@ seed_importer_handle_file (JSContextRef 
 					g_path_get_dirname (file_path), NULL);
     }
 
-  normalized_path = canonicalize_file_name (absolute_path);
+  normalized_path = realpath (absolute_path, NULL);
 
   js_file_dirname = seed_value_from_string (ctx, normalized_path, NULL);
 
--- origsrc/seed-2.27.92/modules/gettext/seed-gettext.c	2009-09-03 15:19:06.000000000 -0500
+++ src/seed-2.27.92/modules/gettext/seed-gettext.c	2009-09-08 01:26:21.600670900 -0500
@@ -312,12 +312,14 @@ seed_module_init(SeedEngine *local_eng)
 	DEFINE_ENUM_MEMBER(ns_ref, LC_MONETARY);
 	DEFINE_ENUM_MEMBER(ns_ref, LC_MESSAGES);
 	DEFINE_ENUM_MEMBER(ns_ref, LC_ALL);
+#ifndef __CYGWIN__
 	DEFINE_ENUM_MEMBER(ns_ref, LC_PAPER);
 	DEFINE_ENUM_MEMBER(ns_ref, LC_NAME);
 	DEFINE_ENUM_MEMBER(ns_ref, LC_ADDRESS);
 	DEFINE_ENUM_MEMBER(ns_ref, LC_TELEPHONE);
 	DEFINE_ENUM_MEMBER(ns_ref, LC_MEASUREMENT);
 	DEFINE_ENUM_MEMBER(ns_ref, LC_IDENTIFICATION);
+#endif
 	DEFINE_ENUM_MEMBER(ns_ref, LC_CTYPE);
 
 	return ns_ref;
--- origsrc/seed-2.27.92/modules/os/seed-os.c	2009-09-03 15:19:06.000000000 -0500
+++ src/seed-2.27.92/modules/os/seed-os.c	2009-09-08 01:13:06.575023000 -0500
@@ -62,7 +62,7 @@ seed_os_realpath (SeedContext ctx,
       EXPECTED_EXCEPTION("os.realpath", "1 argument");
     }
   arg = seed_value_to_string (ctx, arguments[0], exception);
-  ret = canonicalize_file_name(arg);
+  ret = realpath(arg, NULL);
   g_free (arg);
 
   return seed_value_from_string (ctx, ret, exception);
@@ -1108,7 +1108,9 @@ seed_module_init(SeedEngine * eng)
   //  OS_DEFINE_QUICK_ENUM (O_SHLOCK);
   //  OS_DEFINE_QUICK_ENUM (O_EXLOCK);
 
+#if defined (O_ASYNC)
   OS_DEFINE_QUICK_ENUM (O_ASYNC);
+#endif
 #if defined (O_DIRECT)
   OS_DEFINE_QUICK_ENUM (O_DIRECT);
 #endif
