--- origsrc/seed-2.31.91/libseed/Makefile.am	2010-08-30 16:37:38.000000000 -0500
+++ src/seed-2.31.91/libseed/Makefile.am	2010-12-14 23:19:46.612929800 -0600
@@ -36,6 +36,7 @@ libseed_la_CFLAGS = \
 	-DGOBJECT_INTROSPECTION_VERSION=$(GOBJECT_INTROSPECTION_VERSION)
 
 libseed_la_LDFLAGS = \
+	-no-undefined \
 	$(FFI_LDFLAGS)
 
 libseed_la_LIBADD = \
--- origsrc/seed-2.31.91/libseed/seed-importer.c	2010-08-30 16:37:38.000000000 -0500
+++ src/seed-2.31.91/libseed/seed-importer.c	2010-12-14 23:21:09.979879400 -0600
@@ -591,7 +591,7 @@ seed_importer_handle_native_module (JSCo
   GModule *module;
   JSObjectRef module_obj;
   SeedModuleInitCallback init;
-  gchar *file_path = g_strconcat (dir, "/libseed_", prop, ".", G_MODULE_SUFFIX, NULL);
+  gchar *file_path = g_module_build_path(dir, g_strconcat ("seed_", prop, NULL));
 
   SEED_NOTE (IMPORTER, "Trying native module: %s", file_path);
 
--- origsrc/seed-2.31.91/modules/DynamicObject/Makefile.am	2010-08-30 16:37:39.000000000 -0500
+++ src/seed-2.31.91/modules/DynamicObject/Makefile.am	2010-12-15 00:01:28.987045700 -0600
@@ -15,10 +15,14 @@ AM_CPPFLAGS = \
 	$(SEED_PROFILE_CFLAGS)
 
 libseed_DynamicObject_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version -no-undefined \
 	$(GOBJECT_INTROSPECTION_LDFLAGS) \
 	$(SEED_PROFILE_LIBS)
 
+libseed_DynamicObject_la_LIBADD = \
+	$(top_builddir)/libseed/libseed.la \
+	$(GOBJECT_INTROSPECTION_LIBS)
+
 endif
 
 
--- origsrc/seed-2.31.91/modules/cairo/Makefile.am	2010-08-30 16:37:39.000000000 -0500
+++ src/seed-2.31.91/modules/cairo/Makefile.am	2010-12-14 23:18:04.760726800 -0600
@@ -32,7 +32,7 @@ libseed_cairo_la_CFLAGS = \
 	$(SEED_PROFILE_CFLAGS)
 
 libseed_cairo_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version -no-undefined \
 	$(GOBJECT_INTROSPECTION_LDFLAGS) \
 	$(GDK_LDFLAGS) \
 	$(CAIRO_LDFLAGS)
--- origsrc/seed-2.31.91/modules/canvas/Makefile.am	2010-08-30 16:37:39.000000000 -0500
+++ src/seed-2.31.91/modules/canvas/Makefile.am	2010-12-14 23:18:16.565402000 -0600
@@ -17,7 +17,7 @@ libseed_canvas_la_CFLAGS = \
 	$(SEED_PROFILE_CFLAGS)
 
 libseed_canvas_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version -no-undefined \
 	$(GOBJECT_INTROSPECTION_LDFLAGS) \
 	$(CAIRO_LDFLAGS)
 
--- origsrc/seed-2.31.91/modules/dbus/Makefile.am	2010-08-30 16:37:39.000000000 -0500
+++ src/seed-2.31.91/modules/dbus/Makefile.am	2010-12-14 23:18:22.193723900 -0600
@@ -35,7 +35,7 @@ libseed_dbusnative_la_CFLAGS = \
 
 
 libseed_dbusnative_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version -no-undefined \
 	$(GOBJECT_INTROSPECTION_LDFLAGS) \
 	$(DBUS_LDFLAGS) \
 	-L/lib
--- origsrc/seed-2.31.91/modules/example/Makefile.am	2010-08-30 16:37:39.000000000 -0500
+++ src/seed-2.31.91/modules/example/Makefile.am	2010-12-14 23:18:33.147350400 -0600
@@ -15,7 +15,7 @@ AM_CPPFLAGS = \
 	$(SEED_PROFILE_CFLAGS)
 
 libseed_example_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version -no-undefined \
 	$(GOBJECT_INTROSPECTION_LDFLAGS)
 
 libseed_example_la_LIBADD = \
--- origsrc/seed-2.31.91/modules/ffi/Makefile.am	2010-08-30 16:37:39.000000000 -0500
+++ src/seed-2.31.91/modules/ffi/Makefile.am	2010-12-14 23:18:36.996967200 -0600
@@ -23,7 +23,7 @@ libseed_ffi_la_CFLAGS = \
 	$(FFI_CFLAGS)
 
 libseed_ffi_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version -no-undefined \
 	$(GOBJECT_INTROSPECTION_LDFLAGS) \
 	$(FFI_LDFLAGS)
 
--- origsrc/seed-2.31.91/modules/gettext/Makefile.am	2010-08-30 16:37:39.000000000 -0500
+++ src/seed-2.31.91/modules/gettext/Makefile.am	2010-12-14 23:18:42.498675000 -0600
@@ -15,7 +15,7 @@ AM_CPPFLAGS = \
 	$(SEED_PROFILE_CFLAGS)
 
 libseed_gettext_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version -no-undefined \
 	$(GOBJECT_INTROSPECTION_LDFLAGS)
 
 libseed_gettext_la_LIBADD = \
--- origsrc/seed-2.31.91/modules/gtkbuilder/Makefile.am	2010-08-30 16:37:39.000000000 -0500
+++ src/seed-2.31.91/modules/gtkbuilder/Makefile.am	2010-12-14 23:18:46.235888800 -0600
@@ -17,7 +17,7 @@ AM_CPPFLAGS = \
 	$(SEED_PROFILE_CFLAGS)
 
 libseed_gtkbuilder_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version -no-undefined \
 	$(GTK_LDFLAGS) \
 	$(GOBJECT_INTROSPECTION_LDFLAGS)
 
--- origsrc/seed-2.31.91/modules/libxml/Makefile.am	2010-08-30 16:37:39.000000000 -0500
+++ src/seed-2.31.91/modules/libxml/Makefile.am	2010-12-14 23:18:50.426128400 -0600
@@ -18,7 +18,7 @@ AM_CPPFLAGS = \
 	$(SEED_PROFILE_CFLAGS)
 
 libseed_libxml_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version -no-undefined \
 	$(LIBXML_LDFLAGS) \
 	$(GOBJECT_INTROSPECTION_LDFLAGS)
 
--- origsrc/seed-2.31.91/modules/mpfr/Makefile.am	2010-08-30 16:37:39.000000000 -0500
+++ src/seed-2.31.91/modules/mpfr/Makefile.am	2010-12-14 23:18:53.946329800 -0600
@@ -18,7 +18,7 @@ AM_CPPFLAGS = \
 	$(SEED_PROFILE_CFLAGS)
 
 libseed_mpfr_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version -no-undefined \
 	$(GOBJECT_INTROSPECTION_LDFLAGS)
 
 libseed_mpfr_la_LIBADD = -lmpfr -lgmp \
--- origsrc/seed-2.31.91/modules/multiprocessing/Makefile.am	2010-08-30 16:37:39.000000000 -0500
+++ src/seed-2.31.91/modules/multiprocessing/Makefile.am	2010-12-14 23:18:57.742546900 -0600
@@ -17,7 +17,7 @@ libseed_multiprocessing_la_CFLAGS = \
 	$(SEED_PROFILE_CFLAGS)
 
 libseed_multiprocessing_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version -no-undefined \
 	$(GOBJECT_INTROSPECTION_LDFLAGS)
 
 libseed_multiprocessing_la_LIBADD = \
--- origsrc/seed-2.31.91/modules/os/Makefile.am	2010-08-30 16:37:39.000000000 -0500
+++ src/seed-2.31.91/modules/os/Makefile.am	2010-12-14 23:19:01.259748100 -0600
@@ -15,7 +15,7 @@ libseed_os_la_CFLAGS = \
 	$(SEED_PROFILE_CFLAGS)
 
 libseed_os_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version -no-undefined \
 	$(GOBJECT_INTROSPECTION_LDFLAGS) \
 	$(SEED_DEBUG_LDFLAGS)
 
--- origsrc/seed-2.31.91/modules/readline/Makefile.am	2010-08-30 16:37:39.000000000 -0500
+++ src/seed-2.31.91/modules/readline/Makefile.am	2010-12-14 23:19:04.186915500 -0600
@@ -16,7 +16,7 @@ libseed_readline_la_CFLAGS = \
 	$(SEED_PROFILE_CFLAGS)
 
 libseed_readline_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version -no-undefined \
 	$(GOBJECT_INTROSPECTION_LDFLAGS) \
 	$(FFI_LDFLAGS)
 
--- origsrc/seed-2.31.91/modules/sandbox/Makefile.am	2010-08-30 16:37:39.000000000 -0500
+++ src/seed-2.31.91/modules/sandbox/Makefile.am	2010-12-14 23:19:07.854125300 -0600
@@ -14,7 +14,7 @@ AM_CPPFLAGS = \
 	$(SEED_PROFILE_CFLAGS)
 
 libseed_sandbox_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version -no-undefined \
 	$(GOBJECT_INTROSPECTION_LDFLAGS)
 
 libseed_sandbox_la_LIBADD = \
--- origsrc/seed-2.31.91/modules/sqlite/Makefile.am	2010-08-30 16:37:39.000000000 -0500
+++ src/seed-2.31.91/modules/sqlite/Makefile.am	2010-12-14 23:19:11.043105100 -0600
@@ -17,7 +17,7 @@ libseed_sqlite_la_CFLAGS = \
 	$(SEED_PROFILE_CFLAGS)
 
 libseed_sqlite_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version -no-undefined \
 	$(GOBJECT_INTROSPECTION_LDFLAGS) \
 	$(SQLITE_LDFLAGS) \
 	$(SEED_DEBUG_LDFLAGS)
