--- origsrc/seed-2.91.90/libseed/Makefile.am	2011-02-22 16:44:01.000000000 -0600
+++ src/seed-2.91.90/libseed/Makefile.am	2011-03-10 04:30:05.881990500 -0600
@@ -36,6 +36,7 @@ libseed@SEED_GTK_VERSION@_la_CFLAGS = \
 	-DGOBJECT_INTROSPECTION_VERSION=$(GOBJECT_INTROSPECTION_VERSION)
 
 libseed@SEED_GTK_VERSION@_la_LDFLAGS = \
+	-no-undefined \
 	$(FFI_LDFLAGS)
 
 libseed@SEED_GTK_VERSION@_la_LIBADD = \
--- origsrc/seed-2.91.90/modules/DynamicObject/Makefile.am	2011-02-22 16:44:01.000000000 -0600
+++ src/seed-2.91.90/modules/DynamicObject/Makefile.am	2011-03-10 04:29:18.613908100 -0600
@@ -15,10 +15,14 @@ AM_CPPFLAGS = \
 	$(SEED_PROFILE_CFLAGS)
 
 libseed_DynamicObject_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version -no-undefined \
 	$(GOBJECT_INTROSPECTION_LDFLAGS) \
 	$(SEED_PROFILE_LIBS)
 
+libseed_DynamicObject_la_LIBADD = \
+	$(top_builddir)/libseed/libseed.la \
+	$(GOBJECT_INTROSPECTION_LIBS)
+
 endif
 
 
--- origsrc/seed-2.91.90/modules/cairo/Makefile.am	2011-02-22 16:44:01.000000000 -0600
+++ src/seed-2.91.90/modules/cairo/Makefile.am	2011-03-10 04:29:18.629508200 -0600
@@ -32,7 +32,7 @@ libseed_cairo_la_CFLAGS = \
 	$(SEED_PROFILE_CFLAGS)
 
 libseed_cairo_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version -no-undefined \
 	$(GOBJECT_INTROSPECTION_LDFLAGS) \
 	$(GDK_LDFLAGS) \
 	$(CAIRO_LDFLAGS)
--- origsrc/seed-2.91.90/modules/canvas/Makefile.am	2011-02-22 16:44:01.000000000 -0600
+++ src/seed-2.91.90/modules/canvas/Makefile.am	2011-03-10 04:29:18.629508200 -0600
@@ -17,7 +17,7 @@ libseed_canvas_la_CFLAGS = \
 	$(SEED_PROFILE_CFLAGS)
 
 libseed_canvas_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version -no-undefined \
 	$(GOBJECT_INTROSPECTION_LDFLAGS) \
 	$(CAIRO_LDFLAGS)
 
--- origsrc/seed-2.91.90/modules/dbus/Makefile.am	2011-02-22 16:44:01.000000000 -0600
+++ src/seed-2.91.90/modules/dbus/Makefile.am	2011-03-10 04:29:18.629508200 -0600
@@ -35,7 +35,7 @@ libseed_dbusnative_la_CFLAGS = \
 
 
 libseed_dbusnative_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version -no-undefined \
 	$(GOBJECT_INTROSPECTION_LDFLAGS) \
 	$(DBUS_LDFLAGS) \
 	-L/lib
--- origsrc/seed-2.91.90/modules/example/Makefile.am	2011-02-22 16:44:01.000000000 -0600
+++ src/seed-2.91.90/modules/example/Makefile.am	2011-03-10 04:29:18.629508200 -0600
@@ -15,7 +15,7 @@ AM_CPPFLAGS = \
 	$(SEED_PROFILE_CFLAGS)
 
 libseed_example_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version -no-undefined \
 	$(GOBJECT_INTROSPECTION_LDFLAGS)
 
 libseed_example_la_LIBADD = \
--- origsrc/seed-2.91.90/modules/ffi/Makefile.am	2011-02-22 16:44:01.000000000 -0600
+++ src/seed-2.91.90/modules/ffi/Makefile.am	2011-03-10 04:29:18.645108200 -0600
@@ -23,7 +23,7 @@ libseed_ffi_la_CFLAGS = \
 	$(FFI_CFLAGS)
 
 libseed_ffi_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version -no-undefined \
 	$(GOBJECT_INTROSPECTION_LDFLAGS) \
 	$(FFI_LDFLAGS)
 
--- origsrc/seed-2.91.90/modules/gettext/Makefile.am	2011-02-22 16:44:01.000000000 -0600
+++ src/seed-2.91.90/modules/gettext/Makefile.am	2011-03-10 04:29:18.645108200 -0600
@@ -15,7 +15,7 @@ AM_CPPFLAGS = \
 	$(SEED_PROFILE_CFLAGS)
 
 libseed_gettext_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version -no-undefined \
 	$(GOBJECT_INTROSPECTION_LDFLAGS)
 
 libseed_gettext_la_LIBADD = \
--- origsrc/seed-2.91.90/modules/gtkbuilder/Makefile.am	2011-02-22 16:44:01.000000000 -0600
+++ src/seed-2.91.90/modules/gtkbuilder/Makefile.am	2011-03-10 04:29:18.645108200 -0600
@@ -17,7 +17,7 @@ AM_CPPFLAGS = \
 	$(SEED_PROFILE_CFLAGS)
 
 libseed_gtkbuilder_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version -no-undefined \
 	$(GTK_LDFLAGS) \
 	$(GOBJECT_INTROSPECTION_LDFLAGS)
 
--- origsrc/seed-2.91.90/modules/libxml/Makefile.am	2011-02-22 16:44:01.000000000 -0600
+++ src/seed-2.91.90/modules/libxml/Makefile.am	2011-03-10 04:29:18.660708200 -0600
@@ -18,7 +18,7 @@ AM_CPPFLAGS = \
 	$(SEED_PROFILE_CFLAGS)
 
 libseed_libxml_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version -no-undefined \
 	$(LIBXML_LDFLAGS) \
 	$(GOBJECT_INTROSPECTION_LDFLAGS)
 
--- origsrc/seed-2.91.90/modules/mpfr/Makefile.am	2011-02-22 16:44:01.000000000 -0600
+++ src/seed-2.91.90/modules/mpfr/Makefile.am	2011-03-10 04:29:18.660708200 -0600
@@ -18,7 +18,7 @@ AM_CPPFLAGS = \
 	$(SEED_PROFILE_CFLAGS)
 
 libseed_mpfr_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version -no-undefined \
 	$(GOBJECT_INTROSPECTION_LDFLAGS)
 
 libseed_mpfr_la_LIBADD = -lmpfr -lgmp \
--- origsrc/seed-2.91.90/modules/multiprocessing/Makefile.am	2011-02-22 16:44:01.000000000 -0600
+++ src/seed-2.91.90/modules/multiprocessing/Makefile.am	2011-03-10 04:29:18.660708200 -0600
@@ -17,7 +17,7 @@ libseed_multiprocessing_la_CFLAGS = \
 	$(SEED_PROFILE_CFLAGS)
 
 libseed_multiprocessing_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version -no-undefined \
 	$(GOBJECT_INTROSPECTION_LDFLAGS)
 
 libseed_multiprocessing_la_LIBADD = \
--- origsrc/seed-2.91.90/modules/os/Makefile.am	2011-02-22 16:44:01.000000000 -0600
+++ src/seed-2.91.90/modules/os/Makefile.am	2011-03-10 04:29:18.676308200 -0600
@@ -15,7 +15,7 @@ libseed_os_la_CFLAGS = \
 	$(SEED_PROFILE_CFLAGS)
 
 libseed_os_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version -no-undefined \
 	$(GOBJECT_INTROSPECTION_LDFLAGS) \
 	$(SEED_DEBUG_LDFLAGS)
 
--- origsrc/seed-2.91.90/modules/readline/Makefile.am	2011-02-22 16:44:01.000000000 -0600
+++ src/seed-2.91.90/modules/readline/Makefile.am	2011-03-10 04:29:18.676308200 -0600
@@ -16,7 +16,7 @@ libseed_readline_la_CFLAGS = \
 	$(SEED_PROFILE_CFLAGS)
 
 libseed_readline_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version -no-undefined \
 	$(GOBJECT_INTROSPECTION_LDFLAGS) \
 	$(FFI_LDFLAGS)
 
--- origsrc/seed-2.91.90/modules/sandbox/Makefile.am	2011-02-22 16:44:01.000000000 -0600
+++ src/seed-2.91.90/modules/sandbox/Makefile.am	2011-03-10 04:29:18.676308200 -0600
@@ -14,7 +14,7 @@ AM_CPPFLAGS = \
 	$(SEED_PROFILE_CFLAGS)
 
 libseed_sandbox_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version -no-undefined \
 	$(GOBJECT_INTROSPECTION_LDFLAGS)
 
 libseed_sandbox_la_LIBADD = \
--- origsrc/seed-2.91.90/modules/sqlite/Makefile.am	2011-02-22 16:44:01.000000000 -0600
+++ src/seed-2.91.90/modules/sqlite/Makefile.am	2011-03-10 04:29:18.691908300 -0600
@@ -17,7 +17,7 @@ libseed_sqlite_la_CFLAGS = \
 	$(SEED_PROFILE_CFLAGS)
 
 libseed_sqlite_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version -no-undefined \
 	$(GOBJECT_INTROSPECTION_LDFLAGS) \
 	$(SQLITE_LDFLAGS) \
 	$(SEED_DEBUG_LDFLAGS)
